core-test/src/test/scala/org/bitcoins/core/crypto/ECPrivateKeyTest.scala

```diff
@@ -99,4 +100,16 @@ class ECPrivateKeyTest extends BitcoinSUnitTest {
     ECPrivateKey().toString must be("Masked(ECPrivateKeyImpl)")
   }
 
+  it must "successfully negate itself" in {
+    forAll(CryptoGenerators.nonZeroPrivKey) { privKey =>
+      val negPrivKey = privKey.negate
+      val pubKey = privKey.publicKey
+      val negPubKey = negPrivKey.publicKey
+      assert(pubKey.bytes.tail == negPubKey.bytes.tail)
+      assert(pubKey.bytes.head != negPubKey.bytes.head)
+      assert(
+        privKey.fieldElement.add(negPrivKey.fieldElement) == FieldElement.zero)
+    }
```

Comment:
Should there be an assertion for adding the pub keys as well?

Reply:
I don't think that is necessary given that this is a test for the private keys and that the public keys are being tested for relative correctness in the above two assertions